java/ql/test/library-tests/dataflow/collections/Test.java

```diff
@@ -84,7 +84,17 @@ public void run4() {
     p.put("key", tainted);
     sink(p.getProperty("key")); // Flow
     sink(p.getProperty("key", "defaultValue")); // Flow
-    
+
+    Properties clean = new Properties();
+    sink(clean.getProperty("key", tainted)); // Flow
+  }
+
+  public void run5() {
+    Properties p = new Properties();
+    p.setProperty("key", tainted);
```

Comment:
Like this?
```suggestion
    sink(p.setProperty("key", tainted)); // Flow
```

Reply:
I believe it returns the *old* value, so you actually want to store some taint, then store something clean and check that the return value of the clean store shows taint

Reply:
Yes, you're right, `setProperty` returns the *old* value.
Done in https://github.com/github/codeql/pull/9963/commits/8eebe710ce568c49f57c51d958b211e4e0e4fa63